typing/parmatch.ml

```diff
@@ -2000,27 +2000,29 @@ let check_unused pred casel =
 
 let irrefutable pat = le_pat pat omega
 
-(* An inactive pattern is a pattern whose matching needs only
-   trivial computations (tag/equality tests).
-   Patterns containing (lazy _) subpatterns are active. *)
-
-let rec inactive pat = match pat with
-| Tpat_lazy _ ->
-    false
-| Tpat_any | Tpat_var _ | Tpat_constant _ | Tpat_variant (_, None, _) ->
-    true
-| Tpat_tuple ps | Tpat_construct (_, _, ps) | Tpat_array ps ->
-    List.for_all (fun p -> inactive p.pat_desc) ps
-| Tpat_alias (p,_,_) | Tpat_variant (_, Some p, _) ->
-    inactive p.pat_desc
-| Tpat_record (ldps,_) ->
-    List.exists (fun (_, _, p) -> inactive p.pat_desc) ldps
-| Tpat_or (p,q,_) ->
-    inactive p.pat_desc && inactive q.pat_desc
-
-(* A `fluid' pattern is both irrefutable and inactive *)
-
-let fluid pat =  irrefutable pat && inactive pat.pat_desc
+(* An inactive pattern is a pattern which does not perform side-effects. Patterns
```

Comment:
does not perform side-effects: matching against it may be duplicated, erased or delayed without change in observable behavior of the program. (I think that `pure` would also be a fine name, but `inactive` is acceptable.)

Reply:
Since we are discussing the documentation side, I'd rather move the description of the function to the interface.

I'd also avoid the reference to "performing side-effects" (and the notion of "pure"), since this can have different meaning depending on the context.  @gasche explanation seems to correspond closely to what it needed where the function is used, which is good.

Reply:
Ok. I've moved the comment to the `.mli` and used @gasche's description rather than "performing side-effects".